subsys/bluetooth/host/gatt.c

```diff
@@ -2018,8 +2018,8 @@ static struct bt_att_req *gatt_req_alloc(bt_att_func_t func, void *params,
 {
 	struct bt_att_req *req;
 
-	/* Allocate new request */
-	req = bt_att_req_alloc(BT_ATT_TIMEOUT);
+	/* This function is called from asynchronous bt_gatt functions and shall not block/wait. */
+	req = bt_att_req_alloc(K_NO_WAIT);
```

Comment:
```
	if (k_current_get() == &bt_rx_thread) {
		return bt_att_req_alloc(K_NO_WAIT);
	}
```

Reply:
And then add a comment similar to https://github.com/zephyrproject-rtos/zephyr/blob/main/include/bluetooth/gatt.h#L985